### DISCUSSION TITLE
HelTec CubeCell HTCC-AB02S

### DISCUSSION THREAD

**Discussion by javrolijk** [timmbogner/Farm-Data-Relay-System#116] [General]
Hello,

I've just acquired the HelTec CubeCell HTCC-AB02S dev board and was wondering if it's compatible with the FDRS system.
I've included below a snapshot of the pins and the link to the module on GitHub as a reference.
https://github.com/HelTecAutomation/CubeCell-Arduino
![image](https://user-images.githubusercontent.com/70583691/198408269-972260e9-c06d-4aef-a3c3-6d59fb655936.png)

**Comment by timmbogner** [comment]
After much research... I'm not sure! It definitely won't work as a gateway because it is not based on an ESP board.
The manufacturer's examples use their own library, but I see the SPI pins for LoRa so it may be possible to use the generic LoRa library. I've succeeded in compiling a node sketch for the CubeCell framework via Arduino, but I had to bypass the library FDRS uses to get a unique ID. The next challenge would be to plug in the LoRa pin numbers and try it out.

The 'dev' branch is currently left in this debugging state so that you can test out different LoRa pins and see if they work. I'm unclear on what pins to assign LORA_RST and LORA_DIO to. I'm hoping they can be left unused. Make sure to disable USE_ESPNOW.

**Comment by nglessner** [comment]
The HTCC-AB02S is not compatible with FDRS as it currently stands simply because it's using the SX1262 lora chip which is not yet supported.  Implementing it isn't a huge task, but I haven't gotten around to it yet. I have a couple of the HTCC-AB02 boards and two of the Heltec WiFi LoRa 32 (V3) boards (both which use the SX1262) so I will eventually implement it if someone doesn't beat me to it. It should be pretty simple as @timmbogner already [pointed out](https://github.com/timmbogner/Farm-Data-Relay-System/discussions/108#discussioncomment-3653234), it's just a matter of implementing a callback.

**Reply by timmbogner** [reply]
Ope yep good point, thanks @nglessner. I once again confused it with the the SX1268.
Sorry @javrolijk, hope you didn't spend time on that. That chip should be supported within a month or two.

**Reply by javrolijk** [reply]
Thanks @timmbogner
@nglessner have you gotten any luck making your CubeCell modules work on the FDRS already? Otherwise, I'll wait for Timm to put something up. I'm not an avid programmer but can follow instructions quite nicely.

**Reply by nglessner** [reply]
@javrolijk I haven't had time to work on it - unfortunately the real job has me very busy at the moment.  That being said, I did finally receive a couple Heltec LoRa 32 (version 3) boards recently, so that will be my focus when I get a chance to work on it.  It seems pretty straight forward as Timm suggested - it's just a matter of implementing callbacks for send/receive/timeout.  [This is what I've been looking at most recently](https://github.com/Heltec-Aaron-Lee/WiFi_Kit_series/tree/master/esp32/libraries/LoraWan102/src).

**Reply by javrolijk** [reply]
Awesome! Any chance you have is much appreciated.

**Reply by timmbogner** [reply]
@nglessner
I was researching that link you posted, particularly the 'radio' directory, and I came upon the [RadioLib project](https://github.com/jgromes/RadioLib). Is it just me, or would it be perfect for bringing wider radio capabilities to FDRS?

**Reply by javrolijk** [reply]
> @nglessner I was researching that link you posted, particularly the 'radio' directory, and I came upon the [RadioLib project](https://github.com/jgromes/RadioLib). Is it just me, or would it be perfect for bringing wider radio capabilities to FDRS?

Hi Tim,
Not an expert, but a universal radio library would be perfect for brother compatibility. I'd say something to keep investigating.
